src/value.rs

```diff
@@ -90,21 +94,45 @@ impl ValueSum {
     /// This only enforces that the value is a signed 63-bit integer. Callers should
     /// enforce any additional constraints on the value's valid range themselves.
     pub fn from_raw(value: i64) -> Self {
-        ValueSum(value)
+        ValueSum(value as i128)
     }
 }
 
 impl Add for ValueSum {
-    type Output = Result<ValueSum, OverflowError>;
+    type Output = Option<ValueSum>;
 
+    #[allow(clippy::suspicious_arithmetic_impl)]
     fn add(self, rhs: Self) -> Self::Output {
-        self.0.checked_add(rhs.0).map(ValueSum).ok_or(OverflowError)
+        self.0
+            .checked_add(rhs.0)
+            .filter(|v| v > &(-(std::u64::MAX as i128)) && v < &(std::u64::MAX as i128))
+            .map(ValueSum)
     }
 }
 
 impl<'a> Sum<&'a ValueSum> for Result<ValueSum, OverflowError> {
     fn sum<I: Iterator<Item = &'a ValueSum>>(iter: I) -> Self {
-        iter.fold(Ok(ValueSum(0)), |acc, cv| acc? + *cv)
+        iter.fold(Ok(ValueSum(0)), |acc, v| (acc? + *v).ok_or(OverflowError))
+    }
+}
+
+impl Sum<ValueSum> for Result<ValueSum, OverflowError> {
+    fn sum<I: Iterator<Item = ValueSum>>(iter: I) -> Self {
+        iter.fold(Ok(ValueSum(0)), |acc, v| (acc? + v).ok_or(OverflowError))
+    }
+}
+
+impl TryFrom<ValueSum> for i64 {
+    type Error = OverflowError;
+
+    fn try_from(v: ValueSum) -> Result<i64, Self::Error> {
+        i64::try_from(v.0).map_err(|_| OverflowError)
+    }
+}
+
+impl From<NoteValue> for ValueSum {
+    fn from(v: NoteValue) -> Self {
+        ValueSum(v.0 as i128)
     }
 }
```

Comment:
Hmm, when would we want to use this instead of an explicit `NoteValue - NoteValue`? This `From` impl only enables obtaining the `ValueSum` for an `Action` with a dummy (zero-value) output; I'm not sure that's notable enough (compared to other constructs) to warrant being the canonical `value.into()` result.